### PR TITLE
Add metdata resource to AD role + ENV Var adjustments

### PR DESCRIPTION
**Pull request by freecaykes**
## Description

New RBAC Permissions for AD Pods querying AD API and Updates in ENV_VARs 

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/render/intrusion_detection.go
Comment (rene-dekker) [inline pkg/render/intrusion_detection.go:1521]: I think the preferred way would be to actually use the existing function `relasticsearch.ContainerDecorate(...)`. For an example, see compliance.go. We have a convenient function that normalizes all of these fields. Would it be possible to modify the adDetectorsImage, such that these standard fields can be continued to be supported?

File: pkg/render/intrusion_detection.go
Comment (rene-dekker) [inline pkg/render/intrusion_detection.go:1540]: Your pod and container should be using:
```
c.cfg.TrustedCertBundle.Volume(),
c.cfg.TrustedCertBundle.VolumeMount(...),
```
for mounting the certificate bundle, which includes already the `esgwCertificate` that allows you to trust elasticsearch.

File: pkg/render/intrusion_detection.go
Comment (rene-dekker) [reply pkg/render/intrusion_detection.go:1521]: And this:
```
					{
						Name: "es-certs",
						VolumeSource: corev1.VolumeSource{
							Secret: &corev1.SecretVolumeSource{
								SecretName: relasticsearch.PublicCertSecret,
								Items: []corev1.KeyToPath{
									{Key: "tls.crt", Path: "es-ca.pem"},
								},
							},
						},
					},
```
Can then be removed.

File: pkg/render/intrusion_detection.go
Comment (rene-dekker) [reply pkg/render/intrusion_detection.go:1540]: And this:
```
					{
						Name: "es-certs",
						VolumeSource: corev1.VolumeSource{
							Secret: &corev1.SecretVolumeSource{
								SecretName: relasticsearch.PublicCertSecret,
								Items: []corev1.KeyToPath{
									{Key: "tls.crt", Path: "es-ca.pem"},
								},
							},
						},
					},
```
Can then be removed.

Comment (freecaykes) [reply pkg/render/intrusion_detection.go:1540]: https://github.com/tigera/operator/pull/2074/commits/72b90570405d14a3140de0aabb467ac8db22db4a

File: pkg/render/intrusion_detection.go
Comment (freecaykes) [reply pkg/render/intrusion_detection.go]: https://github.com/tigera/operator/pull/2074/commits/72b90570405d14a3140de0aabb467ac8db22db4a - set to ELASTIC_* in AD as well

File: pkg/render/intrusion_detection.go
Comment (rene-dekker) [inline pkg/render/intrusion_detection.go:1449]: I think you could simply add "ADLogTypeMetaDataResourceName" to line 1435 and omit this block.

File: pkg/render/intrusion_detection_test.go
Comment (rene-dekker) [inline pkg/render/intrusion_detection_test.go:414]: Maybe if you fully spell the value here, we never risk changes to   `dns.GetServiceDNSNames()` to break your feature.

File: pkg/render/intrusion_detection.go
Comment (rene-dekker) [inline pkg/render/intrusion_detection.go:1500]: I think it is better to just type it out and avoid  `dns.GetServiceDNSNames()`.

Comment (rene-dekker) [reply pkg/render/intrusion_detection.go:1500]: And perhaps store that value in a const.

**Review by rene-dekker — APPROVED** [review]
lgtm

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Pull request merged**
